### ISSUE TITLE
About the model reproduction results of Resnet

### ISSUE THREAD

**Issue by WEIZHIHONG720** [Michel-liu/GroupPose#7]
hi,As shown in the following figure in your paper: In the 50th epoch, the accuracy of the model suddenly increased and finally converged to 72.0. My replication situation is that in the 50th epoch, the accuracy increased from 67.2 to 69.8 and finally converged to 70.8. May I ask what is the specific training log on your end?Looking forward to your reply。

<img width="658" alt="image" src="https://github.com/Michel-liu/GroupPose/assets/43931281/09ebc103-6619-4555-a5fc-74b2d73401c5">

**Comment by Michel-liu** [comment]
Sorry for the delayed reply. I've been very busy lately. The increase (from 68.8 to 71.1) in AP is due to the learning rate decay.

**Comment by WEIZHIHONG720** [comment]
> Sorry for the delayed reply. I've been very busy lately. The increase (from 68.8 to 71.1) in AP is due to the learning rate decay.

Oh, thank you for your reply. Our reproduction difference is that you have increased from 68.6 to 71.1, while I have increased from 67.2 to 69.8. Is the configuration file for resnet the default configuration? I will reproduce the results again.

**Comment by WEIZHIHONG720** [comment]
> Sorry for the delayed reply. I've been very busy lately. The increase (from 68.8 to 71.1) in AP is due to the learning rate decay. Can you share your full training log? thank you very much!

**Comment by WEIZHIHONG720** [comment]
@Michel-liu  Can you share your full training log? thank you very much!

**Comment by Michel-liu** [comment]
Sorry for the delayed reply. I have been too busy recently. The provided configuration is the same setting as our paper. Do our pre-trained checkpoints run well on your machine? Maybe you can try to enlarge the number of instances? Figure 4 already lists the convergence curve, you can take it as the training records for reference.

**Comment by WEIZHIHONG720** [comment]
> Sorry for the delayed reply. I have been too busy recently. The provided configuration is the same setting as our paper. Do our pre-trained checkpoints run well on your machine? Maybe you can try to enlarge the number of instances? Figure 4 already lists the convergence curve, you can take it as the training records for reference.
hi, I used  your pre-trained checkpoints (swin-L) just got the 74.4 AP. 
 
<img width="615" alt="image" src="https://github.com/Michel-liu/GroupPose/assets/43931281/9590c33d-1666-49eb-ac9e-7c0d058a7ccb">

**Comment by WEIZHIHONG720** [comment]
> Sorry for the delayed reply. I have been too busy recently. The provided configuration is the same setting as our paper. Do our pre-trained checkpoints run well on your machine? Maybe you can try to enlarge the number of instances? Figure 4 already lists the convergence curve, you can take it as the training records for reference.

hi, I used your pre-trained checkpoints (swin-L) just got the 74.4 AP.
<img width="615" alt="image" src="https://github.com/Michel-liu/GroupPose/assets/43931281/4de93e99-62f4-41dd-b5d8-10d89c7db2f5">

**Comment by Michel-liu** [comment]
Okay, I get it. Maybe the environment affects the final results. Before releasing the checkpoint, we have already checked each of them. I will provide a detailed Python environment configuration next month, I am too busy right now, sorry.

**Comment by Michel-liu** [comment]
Sorry for the delayed reply. I have checked the checkpoint (Swin-L on COCO) again. It works normally. I guess it's an environmental issue. The detailed environmental configurations are as follows. Feel free to give me feedback at any time.

[requirements.txt](https://github.com/Michel-liu/GroupPose/files/13439197/requirements.txt)

**Comment by WEIZHIHONG720** [comment]
@Michel-liu  Thanks! I will try it again. By the way, I am working on the interpretability of end-to-end pose estimation models, could I have your contact information? and we can exchange ideas sometime. My e-mail address is weizh588@mail.ustc.edu.cn.

**Comment by rrenle** [comment]
@WEIZHIHONG720 Have you tried it again? I can't reproduce the result of Resnet(AP=70.5), either.By the way, what is your version of cuda that comes with the driver? Does that have any impact on the results?

**Comment by WEIZHIHONG720** [comment]
@Iterfqly  Yes, I can't reproduce the results of the paper either, I got AP 70.6.

**Comment by rrenle** [comment]
@WEIZHIHONG720 According to my results, if the cuda version that comes with the driver is 11.1, the result of Resnet is AP70.5, if the cuda version is changed to 11.0 which is same with the ones of conda enviroment, the result is AP0.0 after 15 epochs. It is very confusing.

**Comment by WEIZHIHONG720** [comment]
@WEIZHIHONG720  My cuda version is 11.1 . But I really cannot reproduce 72.0AP （resnet50）in the paper.

**Comment by WEIZHIHONG720** [comment]
@Iterfqly My cuda version is 11.1 . But I really cannot reproduce 72.0AP （resnet50）in the paper.

**Comment by rrenle** [comment]
@WEIZHIHONG720 My batch size is 16 (8 GPU cards ✖ 2 images on each card)，which is same with original config. And my GPU  is RTX A5000. Can you tell me your GPU model and quantity?

**Comment by Michel-liu** [comment]
The results of ResNet50 are strange. As many issues are reported on this, I will check it out later.

Before that, do the provided checkpoints work normally? Can the reproduced results of Swin-T and Swin-L be aligned?

**Comment by WEIZHIHONG720** [comment]
@Michel-liu The checkpoint you provided is normal. My result of Swin-T is AP 72.79

**Comment by WEIZHIHONG720** [comment]
> @WEIZHIHONG720 My batch size is 16 (8 GPU cards ✖ 2 images on each card)，which is same with original config. And my GPU is RTX A5000. Can you tell me your GPU model and quantity?
My batch size is the same to you . My GPU is RTX 8000

**Comment by WEIZHIHONG720** [comment]
> @WEIZHIHONG720 My batch size is 16 (8 GPU cards ✖ 2 images on each card)，which is same with original config. And my GPU is RTX A5000. Can you tell me your GPU model and quantity?

My batch size is the same to you . My GPU is RTX 8000

**Comment by WEIZHIHONG720** [comment]
@Iterfqly How about your reproduced results of Swin-T ?

**Comment by rrenle** [comment]
> @Iterfqly How about your reproduced results of Swin-T ?

I did not reproduce Swin-T due to memory limitations. Are you unable to reproduce the results of Swin-T, also?

**Comment by tenforests** [comment]
@Michel-liu @WEIZHIHONG720  Hi, first of all, thank you for your fantastic work! But I also can't reproduce the results of resnet50 trained  from scratch, my accuracy is 70.6. do you know what the problem is please? I used 8 * v100 with total batchsize of 16.

**Comment by WEIZHIHONG720** [comment]
> > @Iterfqly How about your reproduced results of Swin-T ?
> 
> I did not reproduce Swin-T due to memory limitations. Are you unable to reproduce the results of Swin-T, also?

Yes, how about you now.

**Comment by rrenle** [comment]
> > > @Iterfqly How about your reproduced results of Swin-T ?
> > 
> > 
> > I did not reproduce Swin-T due to memory limitations. Are you unable to reproduce the results of Swin-T, also?
> 
> Yes, how about you now.

Sorry, during this period, I did not continue to reproduce it.

**Comment by Aritoria** [comment]
@WEIZHIHONG720 @tenforests @Iterfqly  First of all, thank you very much for your discussion. I'm having a similar issue with the reproduction process, have you fixed the reproduction issue with Resnet50? Can the results of 72.0AP be trained on the validation set?

**Comment by guzejungithub** [comment]
> @Iterfqly How about your reproduced results of Swin-T ?



> ###
> **[WEIZHIHONG720](/WEIZHIHONG720) ** commented [Oct 11, 2023](#issue-1937149298) •

Hi, have you sloved this problem?
